psi4/driver/driver.py

```diff
@@ -1693,8 +1739,18 @@ def frequency(name, **kwargs):
     if freq_mode == 'sow':
         return 0.0
 
-    wfn.frequencies().print_out()
-    core.thermo(wfn, wfn.frequencies())
+    # Project final frequencies?
+    translations_projection_sound = not core.get_option('SCF', 'EXTERN') and \
```

Comment:
This seems like a repeat of line 1313.  In the spirit of the DRY principle, it may be a good idea to centralize this logic; being able to add new cases to only one location would be great

Reply:
Yeah, its awkward b/c (1) the former is in the inner fn `hessian` and there's no easy place to return the value to the outer `frequency` fn and (2) they're actually controlling two different projections – in hessian, whether to include rot in SALCs if findif, and in freq, whether to project rot in the vib analysis, whether analytic or findif.

As soon as c-side findif is dead, I agree a unified location (and hat value) is desirable.

Reply:
Regarding your second point, one solution might be to have two helper functions like `energy_is_rotationally_invariant` and `energy_is_translationally_invariant`, which could be called as part of the determination of the different `*_projection_sound` determinations, with appropriate additions to the resulting bool for the findif / hessian projection.  Currently both have the same dependence on external potentials, but something like simple periodic boundary conditions would make the energy translationally, but not rotationally, invariant.  My fear is that we could introduce something like PBC and forget to add it to the list of conditions in one place or another, potentially creating a mess.  I'm sure you've already thought this through, but just wanted to voice my concerns so we keep this stuff in mind through the migration to py-side analysis.